### PR TITLE
kubernetes_master: introduce self-updating imageRepository

### PR DESCRIPTION
**Pull request by sk4zuzu**
This is the fourth PR in series. :+1::innocent:

In this PR we fix switching between custom image registry and other methods by modifying both static pod definitions and kube-system deamonsets, deployments and statefulsets. This may be a little controversial, please review thoroughly. :hugs:

### CODE REVIEW

**Review by rafzei — DISMISSED** [review]
Not tested but its looks ok for me

File: docs/home/howto/CLUSTER.md [hunk 1]
```diff
@@ -166,6 +166,32 @@ To setup the cluster do the following steps:
...
+specification:
+  custom_image_registry_address: "10.50.2.1:5000"
+  custom_repository_url: "http://10.50.2.1:8080/epirepo"
+  use_ha_control_plane: true
+  promote_to_ha: false
+```
+
+The repository and image registry implementation must be compatible with already existing ansible code:
+
+- the repository data (apt or yum packages) is served from Apache2 HTTP server exactly as structured in the offline package
```
Comment (to-bar) [inline docs/home/howto/CLUSTER.md:188]:
```suggestion
- the repository data (including apt or yum repository) is served from HTTP server and structured exactly as in the offline package
```

File: docs/home/howto/CLUSTER.md [hunk 2]
```diff
@@ -166,6 +166,32 @@ To setup the cluster do the following steps:
...
+- the image registry data is loaded into and served from standard docker registry implementation
```
Comment (to-bar) [inline docs/home/howto/CLUSTER.md:189]:
```suggestion
- the image registry data is loaded into and served from standard Docker registry implementation
```

**Review by to-bar — CHANGES_REQUESTED** [review]

**Review by atsikham — APPROVED** [review]
I had questions about usage of `none://` and updating only `kube-system`, clarified them with @to-bar. LGTM
* `spliturl` requires a protocol, so this is just dummy / fake string
* other namespaces that are managed by Epiphany code should be updated during `epicli apply` for example by applications role

File: docs/home/howto/CLUSTER.md [hunk 3]
```diff
@@ -166,6 +166,32 @@ To setup the cluster do the following steps:
...
+
+Custom urls need to be specified inside the `configuration/shared-config` document, for example:
+
+```yaml
+kind: configuration/shared-config
+title: Shared configuration that will be visible to all roles
+name: default
+specification:
+  custom_image_registry_address: "10.50.2.1:5000"
+  custom_repository_url: "http://10.50.2.1:8080/epirepo"
```
[the remark this answers is not in the document]
Comment (sk4zuzu) [reply docs/home/howto/CLUSTER.md:181]: I think this should answer your question https://github.com/epiphany-platform/epiphany/blob/develop/core/src/epicli/data/common/ansible/playbooks/roles/helm/defaults/main.yml#L6 :+1:

By default we use 80 and the tunnel, but if we set custom repo this port can be any number and  it's later used across the whole ansible codebase. :innocent:

[the remark this answers is not in the document]
Comment (to-bar) [reply docs/home/howto/CLUSTER.md:181]: Thanks 👍

**Review by to-bar — APPROVED** [review]

**Review by atsikham — APPROVED** [review]

**Pull request merged**